packages/firestore/src/platform/persistence.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FirebaseNamespace } from '@firebase/app-types';
+import { _FirebaseNamespace } from '@firebase/app-types/private';
+import { clearPersistence, enablePersistence } from '../api/persistence';
+
+/**
+ * Patches the Firestore prototype and provides the backing implementations for
+ * `Firestore.enablePersistence()` and `Firestore.clearPersistence()`.
+ */
+export function registerFirestorePersistence(
```

Comment:
Instead of patching prototype, a cleaner way might be passing the persistence logics as a constructor parameter to `Firestore` class. Something like:
```
class Firestore {
 constructor(
    databaseIdOrApp: FirestoreDatabase | FirebaseApp,
    authProvider: Provider<FirebaseAuthInternalName>,
    private persistence: FirestorePersistence | null
  ) { ... }

  enablePersistence() { 
       if ( this.persistence) {
         ...
       } else {
          throw new Error(PERSISTENCE_MISSING_ERROR_MSG);
       }
  }
}
```
It can potentially allow us to export persistence logics and let developers decide whether or not to use persistence instead of us making multiple build targets.

Reply:
`we can also create Memory-only Firestore class and a PeristenceFirestore class that extends from it` is also a good option.

Reply:
I updated the PR to use two different Firestore classes.